### PR TITLE
Support more aggregations and comments in Dedalus (#254).

### PR DESCRIPTION
**Pull request by davidchuyaya**

### CODE REVIEW

File: hydroflow_datalog_core/src/lib.rs [hunk 1]
```diff
@@ -331,9 +331,33 @@ fn apply_aggregations(
...
-                        }
+                        },
+                        AggregationType::Sum(_) => {
+                            parse_quote!(prev + #val_at_index)
+                        },
+                        AggregationType::Count(_) => {
+                            parse_quote!(prev+1)
+                        },
+                        AggregationType::Choose(_) => {
+                            parse_quote!(prev) // choose = select any 1 element from the relation. By default we select the 1st.
```
Comment (MingweiSamuel) [inline hydroflow_datalog_core/src/lib.rs:347]: this seems a little to deterministic

Comment (davidchuyaya) [reply hydroflow_datalog_core/src/lib.rs:347]: that's ok tho

File: hydroflow_datalog_core/examples/voting/leader.dl [hunk 2]
```diff
@@ -0,0 +1,11 @@
+.input clientIn
```
Comment (shadaj) [inline hydroflow_datalog_core/examples/voting/leader.dl:1]: If these aren't being used in any actual tests we should probably drop them for now.

File: hydroflow_datalog_core/src/lib.rs [hunk 3]
```diff
@@ -590,4 +614,55 @@ mod tests {
...
+
+    #[test]
+    fn test_choose() {
+        test_snapshots!(
+            r#"
+            .input clientIn
+            .output clientOut
+            .input clientAddr
+            
+            clientOut@addr(choose(i)) :~ clientIn(i), clientAddr(addr)
```
Comment (shadaj) [inline hydroflow_datalog_core/src/lib.rs:626]: Can we add runtime tests for this and count? Doesn't need to be a full Dedalus test, we can just run in Datalog + ticks single-node mode.

**Review by shadaj — CHANGES_REQUESTED** [review]

**Comment by MingweiSamuel** [comment]
Looks like some of Shadaj's commits snuck in?

**Comment by davidchuyaya** [comment]
> 

Yeah it's based off `deploy-custom-services` since that was my local branch

**Comment by shadaj** [comment]
@davidchuyaya can you rebase when you get a chance, that branch has been merged in

**Comment by MingweiSamuel** [comment]
Hmm

**Pull request closed without merging**